app/src/parser_special.c

```diff
@@ -72,13 +62,41 @@ parser_error_t parser_getItem_NativeTransfer(ExecutableDeployItem item, parser_c
 
     uint8_t new_displayIdx = displayIdx - item.UI_fixed_items;
 
-    if (new_displayIdx > item.UI_runtime_items || displayIdx < item.UI_fixed_items) {
-        return parser_unexpected_number_items;
-    }
     uint32_t dataLength = 0;
     uint8_t datatype = 255;
 
-    if(!app_mode_expert()){
+    if (new_displayIdx >= item.UI_runtime_items || displayIdx < item.UI_fixed_items) {
+        return parser_unexpected_number_items;
+    }
+
+    // generic
+    if (item.with_generic_args > 0){
+        if (new_displayIdx == 0) {
+            char *name = "native-transfer";
+            uint32_t name_len = strlen(name);
+            // move offset to the end of args
+            CHECK_PARSER_ERR(parseRuntimeArgs(ctx, item.UI_runtime_items));
```

Comment:
How can this work when parseRuntimeArgs returns whenever it sees the first non-supported type?

Reply:
I think it could be causing the difference between the hash in the test vectors and this.
I will check it out..as here the app should have error any unsupported type, there could be more args, in that sense generic.